crypto/keyring/keyring.go

```diff
@@ -217,95 +242,81 @@ func newKeystore(kr keyring.Keyring, opts ...Option) keystore {
 		optionFn(&options)
 	}
 
-	return keystore{kr, options}
+	return keystore{kr, cdc, options}
 }
 
 func (ks keystore) ExportPubKeyArmor(uid string) (string, error) {
-	bz, err := ks.Key(uid)
+	k, err := ks.Key(uid)
 	if err != nil {
 		return "", err
 	}
 
-	if bz == nil {
-		return "", fmt.Errorf("no key to export with name: %s", uid)
+	key, err := k.GetPubKey()
+	if err != nil {
+		return "", err
 	}
 
-	return crypto.ArmorPubKeyBytes(legacy.Cdc.MustMarshal(bz.GetPubKey()), string(bz.GetAlgo())), nil
+	return crypto.ArmorPubKeyBytes(legacy.Cdc.MustMarshal(key), key.Type()), nil
```

Comment:
When exporting pubkeys, we should also use ks.cdc (proto)

Reply:
I believe, this is good way to go:
```
	bz, err := ks.cdc.MarshalInterface(key)
	if err != nil {
		return "", err
	}

	return crypto.ArmorPubKeyBytes(bz, key.Type()), nil
}

```